test/integration/user-management-api/services/proxies_test.rb

```diff
@@ -65,6 +65,28 @@ def test_update_for_service_mesh
     assert_response :success
   end
 
+  def test_update_should_deploy
+    Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(false)
+    params = provider_key_params.merge(proxy: { endpoint: 'https://alaska.wild' })
+    
+    Proxy.update_all(apicast_configuration_driven: false)
+    Proxy.any_instance.expects(:deploy!).once
+    
+    put(admin_api_service_proxy_path(params))
+    assert_response :success
+  end
+  
+  def test_update_for_apiap_should_not_deploy
+    Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(true)
+    params = provider_key_params.merge(proxy: { endpoint: 'https://alaska.wild' })
+
+    Proxy.update_all(apicast_configuration_driven: false)
+    Proxy.any_instance.expects(:deploy!).never
```

Comment:
Better testing that no new proxy config was created. Asserting that a method has never been called is too error-prone.

Reply:
OK, makes sense.